errors/abci.go

```diff
@@ -59,6 +59,16 @@ func abciCode(err error) uint32 {
 	}
 
 	for {
+		if c, ok := err.(interface{ Unwrap() []error }); ok {
+			errs := c.Unwrap()
+			for _, e := range errs {
+				if e != nil {
+					err = e
```

Comment:
I am not sure if "first error" wins is correct. You could also have first `coder` error win. or last...  it would be good to understand the scenario better. I have commented on the issue.
Skipping any nil errors does make sense though.  This could be achieved by using Go stdlib errors.Join as well.